server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueAssigner.java

```diff
@@ -46,37 +53,38 @@
 
   private final ScmInfoRepository scmInfoRepository;
   private final DefaultAssignee defaultAssigne;
+  private final IssueUpdater issueUpdater;
   private final ScmAccountToUser scmAccountToUser;
+  private final IssueChangeContext changeContext;
 
   private String lastCommitAuthor = null;
   private ScmInfo scmChangesets = null;
 
-  public IssueAssigner(ScmInfoRepository scmInfoRepository, ScmAccountToUser scmAccountToUser, DefaultAssignee defaultAssigne) {
+  public IssueAssigner(AnalysisMetadataHolder analysisMetadataHolder, ScmInfoRepository scmInfoRepository, ScmAccountToUser scmAccountToUser, DefaultAssignee defaultAssigne,
+    IssueUpdater issueUpdater) {
     this.scmInfoRepository = scmInfoRepository;
     this.scmAccountToUser = scmAccountToUser;
     this.defaultAssigne = defaultAssigne;
+    this.issueUpdater = issueUpdater;
+    this.changeContext = createScan(new Date(analysisMetadataHolder.getAnalysisDate()));
   }
 
   @Override
   public void onIssue(Component component, DefaultIssue issue) {
-    if (issue.isNew()) {
-      // optimization - do not load SCM data of this component if there are no new issues
-      loadScmChangesetsIfNeeded(component);
-
+    boolean authorHasBeenSet = false;
+    if (issue.authorLogin() == null) {
+      loadScmChangesets(component);
       String scmAuthor = guessScmAuthor(issue);
-      issue.setAuthorLogin(scmAuthor);
-      if (scmAuthor != null) {
-        String assigneeLogin = scmAccountToUser.getNullable(scmAuthor);
-        if (assigneeLogin == null) {
-          issue.setAssignee(defaultAssigne.getLogin());
-        } else {
-          issue.setAssignee(assigneeLogin);
-        }
-      }
+      issueUpdater.setNewAuthor(issue, scmAuthor, changeContext);
```

Comment:
you're right scmAuthor can be null, and in this case issueUpdater. setNewAuthor won't do anything